lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -338,9 +345,43 @@ legacyUtxoWitness xpub bytes = TxWitness $ BL.toStrict $ runPut $ do
     putByteString (unXPub xpub)
     putByteString bytes
 
+-- | Decode the contents of a delegated transaction @Transaction@-message.
+getDelegatedCertificateTransaction
+    :: Hash "Tx"
+    -> Get (PoolId, ChimericAccount, Tx, [TxWitness])
+getDelegatedCertificateTransaction tid = do
+    (tx, wits) <- getGenericTransaction tid
+    accId <- getByteString 32
+    poolId <- getByteString 32
+    pure (PoolId poolId, ChimericAccount accId, tx, wits )
+
+putStakeCertificate
+    :: PoolId
+    -> ChimericAccount
+    -> Put
+putStakeCertificate (PoolId poolId) (ChimericAccount accId) = do
+    putByteString accId
+    putByteString poolId
+
+putStakeDelegationTx
+    :: PoolId
+    -> ChimericAccount
+    -> [(TxIn, Coin)]
+    -> [TxOut]
+    -> [TxWitness]
+    -> Put
+putStakeDelegationTx poolId accId inputs outputs witnesses = do
+    putSignedTx inputs outputs witnesses
+    putStakeCertificate poolId accId
+
 -- | Decode the contents of a @Transaction@-message.
 getTransaction :: Hash "Tx" -> Get (Tx, [TxWitness])
-getTransaction tid = label "getTransaction" $ do
```

Comment:
Let's keep this so we can tell the difference between `getTransaction` and `getDelegatedCertificateTransaction`.

Reply:
So I know you did something for this in the beginning, but with https://github.com/input-output-hk/cardano-wallet/pull/932#discussion_r340121044 I wanted to say I'd prefer the stack-traces:
```
getDelegatedCertificateTransaction
getGenericTransaction
# and
getTransaction
getGenericTransaction
```
to
```
getDelegatedCertificateTransaction
# and
getTransaction
```
if the latter meant passing around the label as a parameter (and probably even regardless of that)